File: src/C#/CreationalPatterns/Builder/Laptops/Builder/LaptopBuilder.cs
using Laptops.Product;
using System;
using System.Collections.Generic;
using System.Text;

namespace Laptops.Builder
{
    public abstract class LaptopBuilder
    {
        public Laptop Laptop { get; set; }

        public abstract void SetModel();

        public abstract void SetCPUSeries();

        public abstract void SetGPUModel();

        public abstract void SetRAMType();

        public abstract void SetRAMSize();

        public abstract void SetDisplayType();

        public abstract void SetSSDType();

        public abstract void SetSSDCapacity();

        public abstract void SetExtras();
    }
}

File: src/Java/CreationalPatterns/Builder/Laptops/src/Builder/LaptopBuilder.java
package Builder;

import Product.Laptop;

public abstract class LaptopBuilder {
    protected Laptop laptop;

    public Laptop getLaptop() {
        return laptop;
    }

    public abstract void setModel();

    public abstract void setCPUSeries();

    public abstract void setGPUModel();

    public abstract void setRAMType();

    public abstract void setRAMSize();

    public abstract void setDisplayType();

    public abstract void setSSDType();

    public abstract void setSSDCapacity();

    public abstract void setExtras();
}
